### PR TITLE
Handle settings from config.

### PR DESCRIPTION
**Pull request by manthey**
Added settings processing from the config file and an internal set_setting utility function.  This can be used to set cherrypy settings.

Addresses issue #532.

### CODE REVIEW

**Comment by ronichoudhury** [comment]
I was mostly through a much simpler PR that would have implemented this change. I'd like to discuss our two PRs in person tomorrow before I merge either one. If my simpler approach satisfies the filed issue, I'd like to use that one, but I think probably some mix of this PR and the one I'm working on will be final.

File: docs/setup.rst [hunk 1]
```diff
@@ -124,9 +126,22 @@ plugins          A list of plugins to load (see :ref:`plugin-config`)
 .. [#unset] That is to say, the option is simply unset by default, the
     equivalent of not mentioning the option at all in a configuration file.
 
-.. [#plugins] This option can *only* appear in the configuration file; there is
+.. [#plugins_config] This option can *only* appear in the configuration file; there is
```
Comment (ronichoudhury) [inline docs/setup.rst]: Why this change?

Comment (manthey) [inline docs/setup.rst]: It gets rid of a document build warning.

On Fri, Jan 22, 2016 at 11:50 AM, Roni Choudhury <notifications@github.com>
wrote:

> In docs/setup.rst
> <https://github.com/Kitware/tangelo/pull/542#discussion_r50560208>:
>
> > @@ -124,9 +126,22 @@ plugins          A list of plugins to load (see :ref:`plugin-config`)
> >  .. [#unset] That is to say, the option is simply unset by default, the
> >      equivalent of not mentioning the option at all in a configuration file.
> >
> > -.. [#plugins] This option can *only* appear in the configuration file; there is
> > +.. [#plugins_config] This option can *only* appear in the configuration file; there is
>
> Why this change?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Kitware/tangelo/pull/542/files#r50560208>.
>



-- 
David Manthey
R&D Engineer
Kitware Inc.
(518) 881-4439

**Comment by ronichoudhury** [comment]
Closing in favor of #543
